### DISCUSSION TITLE
Improvement of proxy

### DISCUSSION THREAD

**Discussion by EnelKyss** [dyc3/steamguard-cli#274] [Ideas]
You should consider adding ability to check current IP in steamguard app to check wether proxy is successfully working and also you should first try to ping the proxy to actually check it is working, then send a response to terminal

**Comment by EnelKyss** [comment]
I have just got this error ERROR -
NetworkFailure: Transport failed to make request: error sending request for url (https://api.steampowered.com/ITwoFactorService/FinalizeAddAuthentica
with -
1: error trying to connect: proxy authentication required
    2: proxy authentication required

**Reply by dyc3** [reply]
There's not much I can do about network failures. Do proxy credentials not go in the URL?

**Reply by EnelKyss** [reply]
doesnt look like, also when I checked activity of that proxy, it didnt have any traffic during adding mobile / 2fa

**Comment by EnelKyss** [comment]
Also very weird that everything until now worked :/

**Comment by EnelKyss** [comment]
Looks like it wasnt using the proxy specified at the command when trying to login to account and set mobile
steamguard --http-proxy <...> setup
